pkg/module/build.go

```diff
@@ -39,58 +31,45 @@ func CreateArchive(fs vfs.FileSystem, path, gitRemote string, def *Definition, i
 
 	ctx := cpi.DefaultContext()
 
-	archive, err := comparch.New(
-		ctx,
-		accessobj.ACC_CREATE, archiveFs,
-		nil,
-		nil,
-		vfs.ModePerm,
-	)
 	if err != nil {
 		return nil, fmt.Errorf("unable to build archive for minimal descriptor: %w", err)
 	}
-
-	cd := archive.GetDescriptor()
-	cd.Metadata.ConfiguredVersion = def.SchemaVersion
-	builtByCLI, err := v1.NewLabel("kyma-project.io/built-by", "cli", v1.WithVersion("v1"))
+	descriptorVersioned, err := ocm.Convert(cd, &ocm.EncodeOptions{SchemaVersion: cd.SchemaVersion()})
+	if err != nil {
+		return nil, fmt.Errorf("unable to build archive for minimal descriptor: %w", err)
+	}
+	data, err := yaml.Marshal(descriptorVersioned)
 	if err != nil {
 		return nil, err
 	}
-
-	cd.Provider = v1.Provider{Name: "kyma-project.io", Labels: v1.Labels{*builtByCLI}}
-
-	if isTargetDirAGitRepo {
-		if err := addSources(cd, def, gitRemote); err != nil {
-			return nil, err
-		}
+	file, err := archiveFs.Create("component-descriptor.yaml")
```

Comment:
Maybe I am wrong, but it looks like it's bypassing the ocm functionality - in the previous version this file was created by the library, isn't it?